### ISSUE TITLE
I am trying to do RandomizedSearchCV on ABOD, but surprisingly it does not run?

### ISSUE THREAD

**Issue by vchouksey** [yzhao062/Pyod#3]
Here is my code:

from pyod.models.abod import ABOD
param_grid = {'neighbours': list(range(1, 5,1)),
              'contamination': np.linspace(0.01, 0.05, 5)}

skf = StratifiedKFold(n_splits=10)
folds = list(skf.split(X.toarray(), y_true))
clf = ABOD()
scoring = make_scorer(precision_score)
search = RandomizedSearchCV(estimator=clf, param_distributions=param_grid, scoring=scoring, cv=folds)
search.fit(X.toarray(), y_true)
y_pred= search.predict(X.toarray())
print('Best parameters:%0.10f' % search.best_params_["contamination"], 'Precision score: %0.3f' % precision_score(y_true, y_pred), 
     'Recall score: %0.3f' % recall_score(y_true, y_pred))

**Comment by yzhao062** [comment]
The result makes sense to me. Just want to bring up some other points to consider:
- ABOD is an unsupervised learning method. If you could do parameter search, you do have the ground truth (labels y) of the training data. **If that is the case, what about treating it as a supervised problem with skewed class (data imbalance)**? I guess if you simply run your data with an XGBOOST classifier, the result may be better than unsupervised approaches, e.g. ABOD.

- You could verify if the percentage of the outliers in your data is closer to 0.05 than 0.01 and 1. If that is true, I think the result is ok.

- Not sure about the version of Pyod you are using. You may want to update to the latest version since it is updated everyday (it is not in production stage yet), which can be done "pip install --upgrade pyod"

Let me know if you have other questions:)

Yue

**Comment by vchouksey** [comment]
I am only considering extremely imbalanced datasets
MinorityClassSize / MajorityClassSize < 0.20
